### DISCUSSION TITLE
ESP8266 TFT Display - Support for Material Design Icons

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#21796] [Q&A] [answered]
I wonder whether ESP8266 TFT Display can support Material Design Icons (https://pictogrammers.github.io/@mdi/font/2.0.46/) kind of graphics using the `DisplayText` command.

![image](https://github.com/user-attachments/assets/5df939f2-a259-4031-8245-f30f7105efa8)

There are references in Home Assistant (HA) discussion groups on using a MDI font like:
`
common/fonts/materialdesignicons-webfont.ttf
`
Wonder whether this is possible with Tasmota on ESP8266.

There are some details about Tasmota's built in Font types, though I haven't been able to understand how it works currently.

https://tasmota.github.io/docs/Displays/#fonts

**Comment by sfromis** [comment]
The point you refer to with the docs link is for the built-in fonts. The more general font support is when loading a "RAM font" from a `.fnt` file which are "are special binary versions of GFX fonts", and needs a build with `USE_RAMFONT`. Some existing files [here](https://github.com/arendst/Tasmota/tree/development/lib/lib_display/BinFonts). Note that ESP8266 has a rather low amount of RAM, meaning that options are limited. Still, it might work if you convert a selection to the format.

You'd have a **lot** better options with ESP32+PSRAM, and HASPmota or LVGL.

**Reply by jkwim** [reply]
Q1: Is there a way to preview .fnt files? Is the format different from a .ttf file?

Q2: Is there a way to convert a .ttf file in to .fnt file?

**Reply by sfromis** [reply]
While I know nothing, a quick Google search lead me to this library where the description includes words like "a command-line tool for converting TTF fonts to Adafruit_GFX header format".
https://github.com/adafruit/Adafruit-GFX-Library

Whether that's even useful, I'm not trying to guess.

**Comment by valdus2** [comment, marked as answer]
You can convert whatever font you like.
Tools you will need:
1) GLCD Font Creator (To see the font range)
2)  [FontConvert](https://github.com/KrisKasprzak/FontConvert) (For converting ttf to h files for importing to script editor)
3) [Script Editor](https://www.dropbox.com/sh/0us18ohui4c3k82/AACcVmpZ4AfpdrWE_MPFGmbma?dl=0) (For converting .h to .fnt files)

For example converting some characters from Roboto Bold ttf font:

In command prompt you run the below command
`FontConvert Roboto_Bold.ttf 24 902 973 > Roboto_Bold24.h`

Explaining:
24 is the font size that you want
902 973 The characters range that you want (from 902 to 973) and you have find them from GLCD Font Creator
Roboto_Bold24.h The name of the file that it will create

After that, you open the created Roboto_Bold24.h file in script editor and you convert it by choosing h file to fnt file.

That's it.

**Reply by jkwim** [reply]
Great will give it a try.
